dynamiq/nodes/converters/csv.py

```diff
@@ -0,0 +1,178 @@
+    and metadata from the specified metadata columns.
+
+    Attributes:
+        name (str): Display name of the node.
+        group (NodeGroup.CONVERTERS): Node group classification.
+        input_schema (type[CSVConverterInputSchema]): Schema for validating input parameters.
+    """
+
+    name: str = "CSV File Converter"
+    group: Literal[NodeGroup.CONVERTERS] = NodeGroup.CONVERTERS
+    input_schema: ClassVar[type[CSVConverterInputSchema]] = CSVConverterInputSchema
+
+    def execute(
+        self, input_data: CSVConverterInputSchema, config: RunnableConfig | None = None, **kwargs
+    ) -> dict[str, list[Any]]:
+        """
+        Executes the CSV conversion process.
+
+        Processes one or more CSV files according to the input configuration,
+        converting each row into a document with specified content and metadata.
+        If some files fail to process but at least one succeeds, logs errors and continues.
+        If all files fail, raises the last encountered error.
+
+        Args:
+            input_data (CSVConverterInputSchema): Validated input parameters for the conversion.
+            config (RunnableConfig | None): Optional runtime configuration.
+            **kwargs: Additional keyword arguments passed to execution callbacks.
+
+        Returns:
+            dict[str, list[Any]]: Dictionary containing the list of processed documents
+                under the 'documents' key.
+
+        Raises:
+            Exception: If there are errors reading or processing the CSV files.
+        """
+        config = ensure_config(config)
+        self.run_on_node_execute_run(config.callbacks, **kwargs)
+
+        all_documents = []
+        last_error = None
+        success_count = 0
+        total_files = len(input_data.file_paths or []) + len(input_data.files or [])
+
+        if input_data.file_paths:
+            for path in input_data.file_paths:
+                try:
+                    with open(path, encoding="utf-8") as csv_file:
+                        reader = csv.DictReader(csv_file, delimiter=input_data.delimiter)
+                        rows = list(reader)
+                        docs = self._process_rows(rows, path, input_data.content_column, input_data.metadata_columns)
+                        all_documents.extend(docs)
+                        success_count += 1
+                except Exception as e:
+                    logger.error(f"Error processing file {path}: {str(e)}")
+                    last_error = e
+
+        if input_data.files:
+            for file_obj in input_data.files:
+                source_name = getattr(file_obj, "name", "in-memory file")
+                try:
+                    if isinstance(file_obj, bytes):
+                        file_obj = BytesIO(file_obj)
+                    content = file_obj.read().decode("utf-8") if isinstance(file_obj, BytesIO) else file_obj
+                    reader = csv.DictReader(StringIO(content), delimiter=input_data.delimiter)
```

Comment:
as we supports just bytes and BytesIO this should work and save some memory usage on some file reading operations. But please check
```suggestion
                    if isinstance(file_obj, bytes):
                        file_obj = BytesIO(file_obj)
                    file_obj = TextIOWrapper(file_obj, "utf-8")
                    reader = csv.DictReader(file_obj delimiter=input_data.delimiter)
```